server/portal/apps/auth/middleware.py

```diff
@@ -18,17 +18,9 @@ def __call__(self, request):
         user = auth.get_user(request)
         if not (request.path in ['/logout/', '/login/']) and user.is_authenticated:
             try:
-                with transaction.atomic():
-                    try:
-                        tapis_oauth = TapisOAuthToken.objects.filter(user=user).select_for_update().get()
-                    except ObjectDoesNotExist:
-                        logger.error(f'Authenticated user {user.username} missing Tapis API Token')
-                        raise
-                    if tapis_oauth.expired:
-                        logger.debug(f'Refreshing apis Token for {user.username} as it has expired')
-                        tapis_oauth.refresh_tokens()
-            except Exception:
-                logger.exception(f'Tapis Token refresh failed. Forcing logout for {user.username}')
+                TapisOAuthToken.objects.filter(user=user).get()
+            except ObjectDoesNotExist:
+                logger.error(f'Authenticated user {user.username} missing Tapis API Token')
                 logout(request)
                 return HttpResponse("Unauthorized", status=401)
         response = self.get_response(request)
```

Comment:
Question - Do we still need something like this in the middleware? Right now it just checks if the user exists and does a logout if no user is found

Reply:
Imo, we don't need this middleware at all - we can instead handle token related verification in the client () method